application/models/TokenDynamic.php

```diff
@@ -1105,4 +1105,30 @@ public function getSurveyId()
     {
         return self::$sid;
     }
+
+    /**
+     * Returns true if the token can be used
+     * @return bool
+     */
+    public function canBeUsed()
+    {
+        return !empty($this->token)
+            && (
+                $this->completed == "N"
+                || empty($this->completed)
+                || $this->survey->oOptions->alloweditaftercompletion == "Y"
+            );
+    }
+
+    public function canBeEmailed()
+    {
+        return !empty($this->token)
+            && $this->emailstatus == "OK"
+            && $this->email
+            && $this->completed == "N"
```

Comment:
`( $this->completed == "N" || empty($this->completed)))`

Same than can be used

Reply:
Yes, there seems to be some relationship. So far, don't want to make further changes until we have clear scenarios and better tests in place